tests/record/test_legacy.py

```diff
@@ -193,6 +193,7 @@ def test_read_log_append_time_v1():
 @pytest.mark.parametrize("magic", [0, 1])
 def test_reader_corrupt_record_v0_v1(magic):
     buffer = _make_compressed_batch(magic)
+    buffer = bytearray(buffer)
```

Comment:
Why does it have to be a bytearray here?

Reply:
Lines 201-205:
```python
195     buffer = _make_compressed_batch(magic)
196     buffer = bytearray(buffer)
197     len_offset = 8
198
199     # If the wrapper of compressed messages has a key it will just be ignored.
200     key_offset = 26 if magic else 18
201     new_buffer = (
202         buffer[:key_offset] +
203         b"\x00\x00\x00\x03123" +  # Insert some KEY into wrapper
204         buffer[key_offset + 4:]  # Ignore the 4 byte -1 value for old KEY==None
205     )
```

Memoryviews don't that sort of direct byte addition. Rather than append `.tobytes()` at the end of the buffer slices, I just cast the entire buffer to a `bytearray` at the beginning of the test.

Reply:
Just shrink the buffer and return it. No need for the memoryview. I don't want a difference between C and Python in returned types.